#162948872 password reset

jamesbeamie opened the pull request:
#### What does this PR do?

- Allows a user to request for a password reset
- Provides the user with a password reset link

#### Description of Task to be completed?

* User should enter their emails to be sent a password reset link 
* User should be directed to a page to enter and confirm their new password
#### How should this be manually tested?
* Click forgot password link on the login modal
![image](https://user-images.githubusercontent.com/42789760/53004923-33525e00-3443-11e9-9bbe-e81076f61bc8.png)


* Enter your registered email
![image](https://user-images.githubusercontent.com/42789760/52998348-1febc680-3434-11e9-9972-54ad5bdc1985.png)


#### Reset password
* Go to your email and click the link sent for resetting the password.
* Enter the new password and confirm it
![image](https://user-images.githubusercontent.com/42789760/52998390-41e54900-3434-11e9-812b-3dc5b0199627.png)






#### What are the relevant pivotal tracker stories?
[#162948872 ](https://www.pivotaltracker.com/story/show/162948872)

Betty-Kebenei commented on src/actions/ForgotPassword.js:
```diff
@@ -0,0 +1,71 @@
+
+    handleSubmit = (e) => {
+      e.preventDefault();
+      const email_data = this.state.email;
```
fix this kindly

Betty-Kebenei replied:
For js underscores are not recommended

Betty-Kebenei commented on src/actions/ResetPassword.js:
```diff
@@ -0,0 +1,98 @@
+  constructor(props) {
+    super(props);
+    this.state = { password: null, confirm_password: null, success: null };
+    this.handleChange = this.handleChange.bind(this);
```
you don't have to bind functions if you have used arrow functions

Deekerubo reviewed (commented):
Add _snapshots_ to .gitignore

Commit:
feature(password rest):Password reset link

- user should be able to request for password reset

- user should be presented with a password reset link

[Delivers #162948872]

Betty-Kebenei commented on source.env:
```diff
@@ -0,0 +1 @@
+BASE_URL="https://ah-alpha.herokuapp.com/"
```
Delete this file in git and locally

Betty-Kebenei commented on src/actions/ForgotPassword.js:
```diff
@@ -0,0 +1,64 @@
+import { Form, Message } from "semantic-ui-react";
```
This is a component, it should be in the components' folder. And create actions for it.

Betty-Kebenei commented on src/actions/ResetPassword.js:
```diff
@@ -0,0 +1,95 @@
+import { Form, Message } from "semantic-ui-react";
```
This too is a component

Betty-Kebenei commented on src/assets/css/index.scss:
```diff
@@ -12,4 +12,3 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
-
```
Add this line back

Betty-Kebenei commented on src/assets/js/main.js:
```diff
@@ -0,0 +1,10 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Delete this file in git and locally

Betty-Kebenei commented on src/components/Home.js:
```diff
@@ -1,8 +1,11 @@
 import React from "react";
```
Wachira had fixed this

Ruiru11 reviewed (approved):
looks Good
